File: exercise/matlab/ex2/predict.m
function p = predict(theta, X)
%PREDICT Predict whether the label is 0 or 1 using learned logistic 
%regression parameters theta
%   p = PREDICT(theta, X) computes the predictions for X using a 
%   threshold at 0.5 (i.e., if sigmoid(theta'*x) >= 0.5, predict 1)

m = size(X, 1); % Number of training examples

% You need to return the following variables correctly
p = zeros(m, 1);

% ====================== YOUR CODE HERE ======================
% Instructions: Complete the following code to make predictions using
%               your learned logistic regression parameters. 
%               You should set p to a vector of 0's and 1's
%



% prob = sigmoid(X * theta);
% p( prob>=0.5 ) = 1;
% p( prob<0.5 ) = 0;

% 上面代码也是可以的

%其实最终就是判断有多少大于0.5,大于0.5的就看做是1,其他是0
p(sigmoid( X * theta) >= 0.5) = 1;


% =========================================================================


end

File: exercise/py/ex3/predict.py
import numpy as np
from sigmoid import *

def predict(theta1, theta2, x):
    # Useful values
    m = x.shape[0]
    num_labels = theta2.shape[0]

    # You need to return the following variable correctly
    p = np.zeros(m)

    # ===================== Your Code Here =====================
    # Instructions : Complete the following code to make predictions using
    #                your learned neural network. You should set p to a
    #                1-D array containing labels between 1 to num_labels.
    #
    x = np.c_[np.ones(m), x]
    tmp = sigmoid( np.matmul(x,theta1.T) )
    tmp = np.c_[np.ones(np.size(tmp,axis=0)), tmp]
    tmp = sigmoid( np.matmul(tmp,theta2.T) )
    p = np.argmax(tmp,axis=1)
    # py index start from 0
    p+=1
    return p


